plugins/linux_library_loader.cpp

```diff
@@ -24,23 +24,51 @@ namespace Dive
 class LinuxLibraryLoader : public IDynamicLibraryLoader
 {
 public:
-    NativeLibraryHandle Load(const std::string& path) override
+    absl::StatusOr<NativeLibraryHandle> Load(const std::string& path) override
     {
-        return dlopen(path.c_str(), RTLD_LAZY | RTLD_GLOBAL);
+        // Clear any existing error state from dlopen.
+        dlerror();
+        NativeLibraryHandle handle = dlopen(path.c_str(), RTLD_LAZY | RTLD_GLOBAL);
+        if (handle == nullptr)
+        {
+            const char* error_msg = dlerror();
+            return absl::Status(absl::StatusCode::kUnknown,
```

Comment:
nit: This could be written more succinctly
```suggestion
            return absl::UnknownError(
```

Reply:
fixed